tests/unit/test_ast.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from unittest import mock
+
+import pytest
+
+from snowflake.snowpark import Session
+from snowflake.snowpark._internal.ast import AstBatch
+
+
+@pytest.fixture
+def fake_session():
+    yield mock.create_autospec(Session)
+
+
+class TestAstBatch:
+    def test_reset_id_gen_variant(self, fake_session):
+        ast_batch = AstBatch(fake_session)
+        # Simulate usage that changes _id_gen
+        ast_batch.assign()
+        ast_batch.assign()
+        assert next(ast_batch._id_gen) != 1  # Ensure _id_gen has changed
+
+        # Reset _id_gen and verify it is zero
+        ast_batch.reset_id_gen()
+        assert next(ast_batch._id_gen) == 1
+
+    def test_assign(self, fake_session):
+        ast_batch = AstBatch(fake_session)
+        stmt = ast_batch.assign()
+        assert stmt is not None
+
+    def test_eval(self, fake_session):
+        ast_batch = AstBatch(fake_session)
+        ast_batch.eval(ast_batch.assign("foo"))
```

Comment:
what to assert?

Reply:
There aren't any particularly interesting properties to test after this step. I can delete the test altogether, since it's already the same functionality that we test in `test_flush`, or I can leave it in place as a smoke test (if it passes, it means nothing is obviously broken).

Reply:
ok, maybe just add comment say this is a smoke test.

Reply:
Done.